### PR TITLE
refactor(sbb-timetable): use GetTripsQuery for types and interfaces

### PR DESCRIPTION
**Pull request by osminaz**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

-

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [ ] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/components/sbb-pearl-chain/sbb-pearl-chain.tsx [hunk 1]
```diff
@@ -93,7 +93,11 @@ export class SbbPearlChain {
   }
 
   public render(): JSX.Element {
-    const rideLegs: PtRideLeg[] = this.legs?.filter((leg) => leg?.__typename === 'PTRideLeg');
+    const rideLegs: PTRideLeg[] = this.legs?.filter(
+      (leg) => leg?.__typename === 'PTRideLeg'
+    ) as PTRideLeg[];
+    console.log(rideLegs);
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain/sbb-pearl-chain.tsx:99]:
```suggestion
```

File: src/global/interfaces/pearl-chain-properties.ts [hunk 2]
```diff
@@ -72,38 +66,338 @@ export interface ScheduledStopPoint {
...
-/** A passenger carrying vehicle journey for one specified operation day */
-export interface ServiceJourney {
-  /** List of ServiceProduct attributes and journey hints */
-  /** Status (realtime changes) to Journey. */
-  serviceAlteration: ServiceAlteration;
-  /** List of stop points */
-  stopPoints: ScheduledStopPoint[];
-  /** List of ServiceProduct attributes and journey hints */
-  notices: Notice[];
+export enum BoardingAlightingAccessibilityEnum {
```
Comment (jeripeierSBB) [inline src/global/interfaces/pearl-chain-properties.ts:69]: we generally prefer string literals over enums

File: src/global/interfaces/pearl-chain-properties.ts [hunk 3]
```diff
@@ -72,38 +66,338 @@ export interface ScheduledStopPoint {
...
+  METRO = 'METRO',
+  PLANE = 'PLANE',
+  SHIP = 'SHIP',
+  TAXI = 'TAXI',
+  TRAIN = 'TRAIN',
+  TRAMWAY = 'TRAMWAY',
+  UNKNOWN = 'UNKNOWN',
+}
+
+export type GetTripsQuery = {
```
Comment (jeripeierSBB) [inline src/global/interfaces/pearl-chain-properties.ts:127]: this looks... not readable :D It's also generating massive documentation output for timetable-row readme. Did you copy that from the backend api?

Comment (kyubisation) [reply src/global/interfaces/pearl-chain-properties.ts:127]: I assume this is generated from OpenApi or something similar? I would MUCH prefer this be split into separate interfaces.

Comment (osminaz) [reply src/global/interfaces/pearl-chain-properties.ts]: Hey guys, I split the definitions. I hope this works better for us :D

I tested it in our frontend. and it works fine

**Review by jeripeierSBB — APPROVED** [review]
LGTM

**Pull request merged**